### DISCUSSION TITLE
Is there a way for the app to read a file with a specific extension that is opened by user?

### DISCUSSION THREAD

**Discussion by lisiur** [tauri-apps/tauri#2948] [Q&A]
If I use tauri to open a file with a specified extention, how can tauri get the path of the file?

**Comment by palerdot** [comment]
> If I use tauri to open a file

Is this possible right now with tauri? That is, if the app is not yet opened, I right click a file in the Finder/explorer, would I get an option in `Open with` to the tauri based app? If I do get that option, as the title suggests, how to read the file? In electron, I think there is an event one can listen to when a file is opened by the electron app from the finder/explorer. From the docs it is not clear, if tauri has such an option. To be clear, this is not about opening a file from within the tauri app. This is about opening a file with a tauri app, when the app itself has not been launched (or maybe launched but in the background).

```
Example: An hypothetical `Tauri VScode` app. The app is not launched yet. I right click on a file.

1) Will there be a `Open with Tauri VSCode` option?
2) If yes, on clicking on it how can we read the path of the file?
```

If someone from tauri team can answer this question it will be great. If it not possible right now, some kind of issue tracking this would be helpful.

**Reply by FabianLars** [reply]
Not possible yet. Tracking issue: https://github.com/tauri-apps/tauri/issues/3736

(For what its worth, it could be implemented in apps directly _on Windows_ without explicit support in Tauri if you know how to use Rust)

**Reply by palerdot** [reply]
Thanks for linking the tracking issue. 

Rust is not the problem. It would be great, if there is some doc explaining a sample snippet on how to achieve this. Right now, I'm not sure if there is an event that we should be listening too etc (in electron, there is an event we could listen and handle this scenario)

**Reply by FabianLars** [reply]
There is no Tauri feature so we don't have docs about it. There is no event to listen to either which is why you currently can't implement this yourself for macOS (macOS support needs changes in tao and tauri).

For a windows solution all you really need to do is to search which registry entries to edit and then use the winreg crate to do so. -> This is also possible by changing the wix template if you'd like to set it at install time.

To handle the file open events you can then use https://github.com/tauri-apps/tauri-plugin-single-instance - on Windows file open events work by calling the executable _again_ with the file paths.

This is all the hints i can get without writing the code for you :D Also, look at the issue i linked above _and_ the issues + PRs linked in that issue - there will be many pointers including helpful code :)
